files/common/usr/bin/get-appliance-version

```diff
@@ -18,6 +18,76 @@
 set -o errexit
 set -o pipefail
 
-zfs get -Hpo value \
+function usage() {
+	echo "Usage: $(basename "$0") [-dMmps]"
+	echo
+	echo "Display the version information for the appliance. By default,"
+	echo "the full version (major, minor, and patch portions) are displayed."
+	echo "The following arguments may be used to refine the version"
+	echo "information:"
+	echo "-d, --debug -- provide the entire version including the jenkins"
+	echo "               information, if it exists."
+	echo "-M, --major -- displays the major version"
+	echo "-m, --minor -- displays the major and minor version"
+	echo "-p, --patch -- displays the major, minor, and patch version"
+	echo "-s, --snapshot -- displays the major, minor, patch, and " \
+		"snapshot information"
+	exit 2
+}
+
+options=$(getopt -l "major,minor,patch,snapshot,debug" -o "Mmpsd" -- "$@")
```

Comment:
Can we support a `-h` and/or `--help` option, that'd print the usage information?

e.g. right now:
```
$ ./get-appliance-version -h
getopt: invalid option -- 'h'
```

Reply:
Sure